docs/data/tree-view/simple-tree-view/customization/GmailTreeView.tsx

```diff
@@ -22,96 +32,130 @@ declare module 'react' {
   }
 }
 
-type StyledTreeItemProps = TreeItemProps & {
+interface StyledTreeItemProps extends TreeItemNextProps {
   bgColor?: string;
   bgColorForDarkMode?: string;
   color?: string;
   colorForDarkMode?: string;
   labelIcon: React.ElementType<SvgIconProps>;
   labelInfo?: string;
-  labelText: string;
-};
+}
 
-const StyledTreeItemRoot = styled(TreeItem)(({ theme }) => ({
+const CustomTreeItemRoot = styled(TreeItemNextRoot)(({ theme }) => ({
```

Comment:
For this one we have several solutions:

1. We keep putting everything in the label, it keeps the demo short but it means that we are telling people to pass a `React.ReactNode` to the label, which only works on `SimpleTreeView`, maybe educating people to always pass a string is better

2. We use `useTreeView` and create wrappers for each element

3. We use `useTreeView` and create wrapper only for the root, we apply the styling like @noraleonte did, based on the classes. This will require to manually set classes to the slots because (unless we do differently than Base UI), the MUI X classes are not applied when using the headless API. See https://github.com/mui/mui-x/pull/11721/files#r1492769672 for context